examples/infix_math_parser.py

```diff
@@ -0,0 +1,190 @@
+"""Defines a recursive parser for parsing mathematical expressions in infix notation.
+
+Supports binary, unary, and variadic operations. These can also be customized
+in the InfixExpressionParser class variables. Utilizes some regex to improve its
+performance.
+
+Examples of parsing:
+
+The expression "f_1 + f_2 - 1e-3" is parsed into [['f_1', '+', 'f_2', '-', 0.001]]
+
+The expression "Max(Ln(x) + Lb(Abs(y)), Ceil(Sqrt(garlic) * 3), (potato ** 2) / 4, Abs(cosmic) + 10)"
+is parsed into
+[['Max', [[['Ln', ['x']], '+', ['Lb', [['Abs', ['y']]]]], ['Ceil', [[['Sqrt', ['garlic']], '*', 3]]], [['potato', '**', 2], '/', 4], [['Abs', ['cosmic']], '+', 10]]]]
+"""
+
+from operator import or_
+from typing import ClassVar
+
+from pyparsing import (
+    Forward,
+    Group,
+    Literal,
+    ParserElement,
+    Suppress,
+    delimitedList,
+    infixNotation,
```

Comment:
Thanks for using the new PEP8 names where possible, you missed a couple (`infix_notation`, `DelimitedList`).